bin/download_proteins_entrez.py

```diff
@@ -130,21 +130,23 @@ def main(args=None):
     for taxid_input, microbiomeId in zip(args.taxid_input, args.microbiome_ids):
         reader = csv.DictReader(taxid_input)
         for row in reader:
-            taxIds.append(row["taxon_id"])
-            try:
+            # If the abundance is not defined in the taxid_input it is assigned as 1.
+            if "taxon_id" in row.keys() and "abundance" in row.keys():
+                taxIds.append(row["taxon_id"])
                 print(
                     row["taxon_id"],
                     microbiomeId,
                     row["abundance"],
                     sep="\t",
                     file=args.microbiomes_entities,
                     flush=True,
-                )
-            except KeyError:
-                try:
-                    print(row["taxon_id"], microbiomeId, 1, sep="\t", file=args.microbiomes_entities, flush=True)
-                except KeyError:
-                    sys.exit(f"The format of the input file '{taxid_input.name}' is invalid!")
+                    )
+            elif not "taxon_id" in row.keys() or len(row.keys())>=2:
+                sys.exit(f"The format of the input file '{taxid_input.name}' is invalid!" +
+                         "It needs to be a csv file containing taxon_id,abundance or just taxon_id.")
+            elif "taxon_id" in row.keys():
+                taxIds.append(row["taxon_id"])
+                print(row["taxon_id"], microbiomeId, 1, sep="\t", file=args.microbiomes_entities, flush=True)
```

Comment:
- in case 1: shouldn't here be checked if there are exactly 2 keys, or maybe just if the set of keys equals {"taxon_id", "abundance"}?
- could we list the error case last? (just if you don't mind, somehow I am confused by the order :D)

Reply:
True! I changed the checks to compare between specific set and row.keys() to throw the error whenever none of the two cases (taxon_id or taxon_id,abundance) are used, as this is now independent of the order I could list the error case last!